features/profiles/profile-me.tsx

```diff
@@ -256,33 +257,46 @@ const EditableProfileContactCardAvatar = memo(
     const { addPFP, asset, isUploading } = useAddPfp();
     const profileMeStore = useProfileMeStore(inboxId);
     const { data: profile } = useProfileQuery({ xmtpId: inboxId });
-    const [displayUri, setDisplayUri] = useState<string | undefined>(
-      profileMeStore.getState().avatarUri ?? profile?.avatar ?? undefined
-    );
-
-    // Keep display URI in sync to prevent flashing
-    useEffect(() => {
-      if (asset?.uri) {
-        setDisplayUri(asset.uri);
-      } else {
-        setDisplayUri(profileMeStore.getState().avatarUri ?? profile?.avatar ?? undefined);
+    
+    // Determine which avatar to display with priority: store avatar > profile avatar
+    const storeAvatar = profileMeStore.getState().avatarUri;
+    const profileAvatar = profile?.avatar;
+    
+    // Create a display URI with a cache-busting parameter
+    const getDisplayUri = useCallback(() => {
+      // Priority: local asset (during upload) > store avatar > profile avatar
+      const sourceUri = asset?.uri || storeAvatar || profileAvatar;
+      
+      if (!sourceUri) {
+        return undefined;
       }
-    }, [asset?.uri, profile?.avatar, profileMeStore]);
+      
+      return sourceUri;
+    }, [asset?.uri, storeAvatar, profileAvatar]);
```

Comment:
this seems overkill. Espacially the usecallback, why not just  `const sourceUri = asset?.uri || storeAvatar || profileAvatar;` and below avatarUri=...